pkg/structs/flamebearer/flamebearer.go

```diff
@@ -100,3 +102,46 @@ func newTimeline(timeline *segment.Timeline) *FlamebearerTimelineV1 {
 		Watermarks:    timeline.Watermarks,
 	}
 }
+
+func (fb FlamebearerProfile) Validate() error {
+	if fb.Version > 1 {
+		return fmt.Errorf("unsupported version %d", fb.Version)
+	}
+	return fb.FlamebearerProfileV1.Validate()
+}
+
+// Validate the V1 profile.
+// A custom validation is used as the constraints are hard to define in a generic way
+// (e.g. using https://github.com/go-playground/validator)
+func (fb FlamebearerProfileV1) Validate() error {
+	format := tree.Format(fb.Metadata.Format)
+	if format != tree.FormatSingle && format != tree.FormatDouble {
+		return fmt.Errorf("unsupported format %s", format)
+	}
+	if len(fb.Flamebearer.Names) == 0 {
+		return fmt.Errorf("a profile must have at least one symbol name")
+	}
+	if len(fb.Flamebearer.Levels) == 0 {
+		return fmt.Errorf("a profile must have at least one profiling level")
+	}
+	var mod int
+	switch format {
+	case tree.FormatSingle:
+		mod = 4
+	case tree.FormatDouble:
+		mod = 7
```

Comment:
Maybe we also need to check `leftTicks` and `rightTicks` exist?
https://github.com/pyroscope-io/pyroscope/issues/762

Also `numTicks` and `maxSelf` ?

Reply:
I think I'll leave this out of this PR:
- Having `leftTicks` and `rightTicks` in a single profile is something that shouldn't happen, but it's not a problem either (they'll just be ignored). I think it may be better to focus on a new data format where these requirements are better defined (mandatory when needed, non-existent when they are not).
- I'd prefer to improve our test-suite at tree generation before checking that `numTicks` and `maxSelf` match the expected result.